plugins/sla_check.py

```diff
@@ -0,0 +1,179 @@
+"""
+This provides a way for a DataDog monitor to ping airflow and query for jobs currently
+out of compliance with their statically configured SLA. DAG's can be configured for
+SLA monitoring by adding a DAG_SLA field to them. On access, DAG's are loaded into the
+current runtime and checked for SLA's meaning they can be add/changed without requiring
+a restart of the service.
+
+DataDogs should be configured similar to the following:
+
+init_config:
+
+instances:
+  - name: Airflow SLA's
+    url: http://localhost/admin/slacheckview/healthcheck
+    timeout: 10
+
+    collect_response_time: true
+    include_content: true
+"""
+import imp
+import logging
+import os
+import re
+from datetime import datetime, date, timedelta
+
+from flask import Blueprint, jsonify
+from flask import request
+from flask_admin import BaseView, expose
+
+from airflow import settings
+from airflow.models import DagBag, DagRun
+from airflow.utils.state import State
+
+
+regex = re.compile(r'((?P<hours>\d+?)hr)?((?P<minutes>\d+?)m)?((?P<seconds>\d+?)s)?')
+
+SLACheckBlueprint = Blueprint(
+    "sla_check", __name__,
+    template_folder='templates',
+    static_folder='static',
+    static_url_path='/static/sla_check')
+
+
+# Convenience method for creating the Flask response
+def get_base_response(status="OK", call_time=datetime.now(), include_arguments=True):
+  base_response = {"status": status, "call_time": call_time}
+  if include_arguments:
+    base_response["arguments"] = request.args
+  return base_response
+
+# Convenience method for finishing the Flask response
+def get_final_response(base_response, output=None, airflow_cmd=None):
+  final_response = base_response
+  final_response["response_time"] = datetime.now()
+  if output:
+    final_response["output"] = output
+  if airflow_cmd:
+    final_response["airflow_cmd"] = airflow_cmd
+  return jsonify(final_response)
+
+# Attempts to load a module from a Python file
+def load_dag_module(module_file):
+  module_folder = os.path.dirname(module_file)
+  module_file = os.path.basename(module_file)
+  module_name = os.path.splitext(module_file)[0]
+
+  fp, pathname, description = imp.find_module(module_name, [module_folder])
+
+  try:
+    return imp.load_module(module_name, fp, pathname, description)
+  except ImportError:
+    logging.error("Failed to load module %s", fp)
+  finally:
+    # Since we may exit via an exception, close fp explicitly.
+    if fp:
+      fp.close()
+
+# Loads all of the existing dags in Airflow and compiles a list
+# of all the dag names and sla values for DAG's that have a
+# DAG_SLA defined. DAG modules that cannot be loaded will be skipped.
+def find_slas():
+  dagbag = DagBag()
+  parent_dags = filter(lambda d: d.parent_dag is None, dagbag.dags.values())
+
+  output = []
+  for dag in parent_dags:
+    dag_module = load_dag_module(dag.full_filepath)
+
+    if dag_module and hasattr(dag_module, 'DAG_SLA'):
```

Comment:
this tracks the concept of an SLA as an attribute in a module.  But the SLA is really on the DAG (there might be more than one dag in the same module, they might have different names).  I'm not totally following how this is supposed to work.